src/net/net.c

```diff
@@ -196,11 +196,22 @@ static boolean get_static_config(tuple root, struct netif *n, boolean trace) {
     ip4_addr_t netmask;
     ip4_addr_t gw;
 
-    if (!get_config_addr(root, sym(ipaddr), &ip) ||
-        !get_config_addr(root, sym(gateway), &gw) ||
-        !get_config_addr(root, sym(netmask), &netmask))
+    if (!get_config_addr(root, sym(ipaddr), &ip))
         return false;
 
+    if (!get_config_addr(root, sym(netmask), &netmask))
+        ip4_addr_set_u32(&netmask, lwip_htonl(0xffffff00)); // 255.255.255.0
+
+    if (!get_config_addr(root, sym(gateway), &gw)) {
+        // common best practices are: network + 1 or broadcast - 1,
+        // so we will use latter if former is in use.
+        u32_t ip_after_network = (netmask.addr & ip.addr) + 1;
```

Comment:
I think you still need lwip_ntohl around (netmask.addr & ip.addr) here (and also (~(netmask.addr) | ip.addr) below) before applying any computation, given that the addresses are in network byte order.

Reply:
I have simplified it so we don't need to convert it twice.